vm/actor/src/builtin/market/mod.rs

```diff
@@ -813,17 +817,14 @@ impl Actor {
 
                     let (slash_amount, next_epoch, remove_deal) =
                         msm.update_pending_deal_state(&state, &deal, curr_epoch)?;
-                    assert_ne!(
-                        slash_amount.sign(),
-                        Sign::Minus,
-                        "next scheduled epoch should be undefined as deal has been removed"
-                    );
+                    if slash_amount.sign() == Sign::Minus {
```

Comment:
if you bring in the `Signed` trait from `num_traits` or whatever it's called, you can just call `.is_negative()` and it's a bit easier to read